src/e2e/specs/dashboard.spec.ts

```diff
@@ -328,68 +327,50 @@ test.describe(`${process.env.E2E_TEST_ENV} - Breaches Dashboard - Footer`, () =>
 
     const clickOnATagCheckDomain = async (
       aTag: Locator,
-      expectedStrings: string[],
-    ): Promise<boolean> => {
+      host: string,
+      pathParts: string[],
+    ) => {
       const href = await aTag.getAttribute("href");
       if (href === null) return false;
 
-      const target = await aTag.getAttribute("target");
-
-      if (target === "_blank") {
-        //opens link in a new tab
-        const newPage = await context.newPage();
-        await newPage.goto(href);
-        const currentUrl = newPage.url();
-        await newPage.close();
-        return expectedStrings.every((expectedString) =>
-          currentUrl.includes(expectedString),
-        );
-      } else {
-        //opens link in the same tab
-        await page.goto(href);
-        const currentUrl = page.url();
-        await page.goBack();
-        return expectedStrings.every((expectedString) =>
-          currentUrl.includes(expectedString),
-        );
-      }
+      await page.goto(href);
+      const currentUrl = new URL(page.url());
+      const perceivedHost = currentUrl.hostname;
+      const perceivedPath = currentUrl.pathname;
+      expect(perceivedHost).toBe(host);
+      expect(pathParts.every((part) => perceivedPath.includes(part)));
+      await page.goBack();
     };
+
     await dashboardPage.goToDashboard();
-    expect(page.locator("footer a >> img")).toBeTruthy();
-    expect(
-      await clickOnATagCheckDomain(dashboardPage.mozillaLogoFooter, [
-        "mozilla.org",
-      ]),
-    ).toBeTruthy();
-    expect(
-      await clickOnATagCheckDomain(dashboardPage.allBreachesFooter, [
-        "/breaches",
-      ]),
-    ).toBeTruthy();
-    expect(
-      await clickOnATagCheckDomain(dashboardPage.faqsFooter, [
-        "support.mozilla.org",
-        "/kb/",
-        "firefox-monitor-faq",
-      ]),
-    ).toBeTruthy();
-    expect(
-      await clickOnATagCheckDomain(dashboardPage.termsOfServiceFooter, [
-        "mozilla.org",
-        "/about/legal/terms/subscription-services/",
-      ]),
-    ).toBeTruthy();
-    expect(
-      await clickOnATagCheckDomain(dashboardPage.privacyNoticeFooter, [
-        "mozilla.org",
-        "/privacy/subscription-services/",
-      ]),
-    ).toBeTruthy();
-    expect(
-      await clickOnATagCheckDomain(dashboardPage.githubFooter, [
-        "github.com",
-        "/mozilla/blurts-server",
-      ]),
-    ).toBeTruthy();
+    await expect(page.locator("footer a >> img")).toBeVisible();
+    await clickOnATagCheckDomain(
+      dashboardPage.mozillaLogoFooter,
+      "www.mozilla.org",
+      [],
+    );
+    await clickOnATagCheckDomain(
+      dashboardPage.allBreachesFooter,
+      "stage.firefoxmonitor.nonprod.cloudops.mozgcp.net",
```

Comment:
Although see the other option of possibly using the [`.toHaveURL()` page assertion](https://playwright.dev/docs/api/class-pageassertions#page-assertions-to-have-url), which supports RegExp or string matches -- which might let you be as strict/loose as you want with host/pathname matching.

```js
await expect(page).toHaveURL(/.*checkout/);
```

Reply:
OK, I think I found it. https://github.com/mozilla/blurts-server/blob/75c361ae6a7c316754bf26d455970761d606937e/src/e2e/utils/helpers.ts#L42C16-L43 seems to inject a `process.env.E2E_TEST_BASE_URL` which might be a environment-specific base URL. So it might include that `"https://stage.firefoxmonitor.nonprod.cloudops.mozgcp.net"` string for us. And now that we have a rough idea of what the ENV var is, we can try and find some existing usages in the code using <kbd>git grep</kbd>:

```sh
git grep -n "E2E_TEST_BASE_URL" | cat

.env-dist:118:E2E_TEST_BASE_URL=
.github/workflows/e2e_cron.yml:55:          E2E_TEST_BASE_URL: ${{ secrets.E2E_TEST_BASE_URL }}
.github/workflows/e2e_pr.yml:71:          E2E_TEST_BASE_URL: ${{ secrets.E2E_TEST_BASE_URL }}
playwright.config.js:68:    baseURL: process.env.E2E_TEST_BASE_URL ?? 'https://stage.firefoxmonitor.nonprod.cloudops.mozgcp.net',
src/e2e/global-setup.ts:28:  await page.goto(process.env.E2E_TEST_BASE_URL as string);
src/e2e/pages/landingPage.ts:223:    await this.page.goto(process.env.E2E_TEST_BASE_URL as string);
src/e2e/specs/auth.spec.ts:34:    expect(page.url()).toBe(`${process.env.E2E_TEST_BASE_URL}${successUrl}`);
src/e2e/specs/dashboard.spec.ts:294:    await page.goto(`${process.env.E2E_TEST_BASE_URL}`);
src/e2e/specs/landing.spec.ts:201:    const successUrl = process.env.E2E_TEST_BASE_URL + "/user/welcome";
src/e2e/specs/landing.spec.ts:228:    const successUrl = process.env.E2E_TEST_BASE_URL + "/user/welcome";
src/e2e/specs/landing.spec.ts:262:      process.env.E2E_TEST_BASE_URL +
src/e2e/specs/landing.spec.ts:289:      process.env.E2E_TEST_BASE_URL +
src/e2e/utils/helpers.ts:42:  process.env["E2E_TEST_BASE_URL"] =
```